> **Failed tests: HeaderConsistency**

```
RecoMET/METAlgorithms/interface/HcalHaloAlgo.h:42:47: error: expected ')' before 'iC'
   explicit HcalHaloAlgo(edm::ConsumesCollector iC);
```
looks like an include is missing

--- a/RecoMET/METAlgorithms/interface/EcalHaloAlgo.h
+++ b/RecoMET/METAlgorithms/interface/EcalHaloAlgo.h
@@ -44,7 +44,7 @@
 class EcalHaloAlgo {
 public:
   // constructor
-  EcalHaloAlgo();
+  explicit EcalHaloAlgo(edm::ConsumesCollector iC);
   // destructor
   ~EcalHaloAlgo() {}
 
@@ -117,6 +117,7 @@ private:
   float SumEnergyThreshold;
   int NHitsThreshold;
 
+  edm::ESGetToken<CaloGeometry, CaloGeometryRecord> geoToken_;
   const CaloGeometry* geo;
   math::XYZPoint getPosition(const DetId& id, reco::Vertex::Point vtx);
 };

--- a/RecoMET/METAlgorithms/interface/HcalHaloAlgo.h
+++ b/RecoMET/METAlgorithms/interface/HcalHaloAlgo.h
@@ -36,12 +36,12 @@
 
 #include "DataFormats/METReco/interface/HaloClusterCandidateHCAL.h"
 
+#include "FWCore/Framework/interface/FrameworkfwdMostUsed.h"
+
 class HcalHaloAlgo {
 public:
   //constructor
-  HcalHaloAlgo();
-  //destructor
-  ~HcalHaloAlgo() {}
+  explicit HcalHaloAlgo(edm::ConsumesCollector iC);
 
   // run algorithm
   reco::HcalHaloData Calculate(const CaloGeometry& TheCaloGeometry,
@@ -99,6 +99,7 @@ private:
   float SumEnergyThreshold;
   int NHitsThreshold;
 
+  edm::ESGetToken<CaloGeometry, CaloGeometryRecord> geoToken_;
   const CaloGeometry* geo_;
   const HcalGeometry* hgeo_;
   math::XYZPoint getPosition(const DetId& id, reco::Vertex::Point vtx);

--- a/RecoMET/METAlgorithms/src/EcalHaloAlgo.cc
+++ b/RecoMET/METAlgorithms/src/EcalHaloAlgo.cc
@@ -1,3 +1,4 @@
+#include "FWCore/Framework/interface/ConsumesCollector.h"
 #include "RecoMET/METAlgorithms/interface/EcalHaloAlgo.h"
 #include "DataFormats/Common/interface/ValueMap.h"
 
@@ -16,7 +17,7 @@ using namespace edm;
 
 bool CompareTime(const EcalRecHit* x, const EcalRecHit* y) { return x->time() < y->time(); }
 
-EcalHaloAlgo::EcalHaloAlgo() {
+EcalHaloAlgo::EcalHaloAlgo(edm::ConsumesCollector iC) : geoToken_(iC.esConsumes()) {
   RoundnessCut = 0;
   AngleCut = 0;
   EBRecHitEnergyThreshold = 0.;
@@ -182,9 +183,7 @@ EcalHaloData EcalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeometry,
     TheAngleFiller.fill();
   }
 
-  edm::ESHandle<CaloGeometry> pGeo;
-  TheSetup.get<CaloGeometryRecord>().get(pGeo);
-  geo = pGeo.product();
+  geo = &TheSetup.getData(geoToken_);
 
   //Halo cluster building:
   //Various clusters are built, depending on the subdetector.

--- a/RecoMET/METAlgorithms/src/HcalHaloAlgo.cc
+++ b/RecoMET/METAlgorithms/src/HcalHaloAlgo.cc
@@ -1,3 +1,4 @@
+#include "FWCore/Framework/interface/ConsumesCollector.h"
 #include "RecoMET/METAlgorithms/interface/HcalHaloAlgo.h"
 #include <map>
 
@@ -20,7 +21,7 @@ bool CompareTowers(const CaloTower* x, const CaloTower* y) {
   return x->iphi() * 1000 + x->ieta() < y->iphi() * 1000 + y->ieta();
 }
 
-HcalHaloAlgo::HcalHaloAlgo() : geo_(nullptr), hgeo_(nullptr) {
+HcalHaloAlgo::HcalHaloAlgo(edm::ConsumesCollector iC) : geoToken_(iC.esConsumes()), geo_(nullptr), hgeo_(nullptr) {
   HBRecHitEnergyThreshold = 0.;
   HERecHitEnergyThreshold = 0.;
   SumEnergyThreshold = 0.;
@@ -223,9 +224,7 @@ HcalHaloData HcalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeometry,
     prevHadEt = tower->hadEt();
   }
 
-  edm::ESHandle<CaloGeometry> pGeo;
-  TheSetup.get<CaloGeometryRecord>().get(pGeo);
-  geo_ = pGeo.product();
+  geo_ = &TheSetup.getData(geoToken_);
   hgeo_ = dynamic_cast<const HcalGeometry*>(geo_->getSubdetectorGeometry(DetId::Hcal, 1));
 
   //Halo cluster building:

--- a/RecoMET/METProducers/interface/EcalHaloDataProducer.h
+++ b/RecoMET/METProducers/interface/EcalHaloDataProducer.h
@@ -116,6 +116,7 @@ namespace reco {
     edm::EDGetTokenT<reco::SuperClusterCollection> supercluster_token_;
     edm::EDGetTokenT<reco::PhotonCollection> photon_token_;
     edm::ESGetToken<CaloGeometry, CaloGeometryRecord> calogeometry_token_;
+    EcalHaloAlgo EcalAlgo;
 
     float EBRecHitEnergyThreshold;
     float EERecHitEnergyThreshold;

--- a/RecoMET/METProducers/interface/HcalHaloDataProducer.h
+++ b/RecoMET/METProducers/interface/HcalHaloDataProducer.h
@@ -105,6 +105,7 @@ namespace reco {
     edm::EDGetTokenT<HFRecHitCollection> hfrechit_token_;
     edm::EDGetTokenT<CaloTowerCollection> calotower_token_;
     edm::ESGetToken<CaloGeometry, CaloGeometryRecord> calogeometry_token_;
+    HcalHaloAlgo HcalAlgo;
 
     float HBRecHitEnergyThreshold;
     float HERecHitEnergyThreshold;

--- a/RecoMET/METProducers/src/EcalHaloDataProducer.cc
+++ b/RecoMET/METProducers/src/EcalHaloDataProducer.cc
@@ -12,7 +12,7 @@ using namespace edm;
 using namespace std;
 using namespace reco;
 
-EcalHaloDataProducer::EcalHaloDataProducer(const edm::ParameterSet& iConfig) {
+EcalHaloDataProducer::EcalHaloDataProducer(const edm::ParameterSet& iConfig) : EcalAlgo(consumesCollector()) {
   //RecHit Level
   IT_EBRecHit = iConfig.getParameter<edm::InputTag>("EBRecHitLabel");
   IT_EERecHit = iConfig.getParameter<edm::InputTag>("EERecHitLabel");
@@ -82,7 +82,6 @@ void EcalHaloDataProducer::produce(Event& iEvent, const EventSetup& iSetup) {
   iEvent.getByToken(photon_token_, ThePhotons);
 
   //Run the EcalHaloAlgo to reconstruct the EcalHaloData object
-  EcalHaloAlgo EcalAlgo;
   EcalAlgo.SetRoundnessCut(RoundnessCut);
   EcalAlgo.SetAngleCut(AngleCut);
   EcalAlgo.SetRecHitEnergyThresholds(EBRecHitEnergyThreshold, EERecHitEnergyThreshold, ESRecHitEnergyThreshold);

--- a/RecoMET/METProducers/src/HcalHaloDataProducer.cc
+++ b/RecoMET/METProducers/src/HcalHaloDataProducer.cc
@@ -12,7 +12,7 @@ using namespace edm;
 using namespace std;
 using namespace reco;
 
-HcalHaloDataProducer::HcalHaloDataProducer(const edm::ParameterSet& iConfig) {
+HcalHaloDataProducer::HcalHaloDataProducer(const edm::ParameterSet& iConfig) : HcalAlgo(consumesCollector()) {
   //RecHit Level
   IT_EBRecHit = iConfig.getParameter<edm::InputTag>("EBRecHitLabel");
   IT_EERecHit = iConfig.getParameter<edm::InputTag>("EERecHitLabel");
@@ -63,7 +63,6 @@ void HcalHaloDataProducer::produce(Event& iEvent, const EventSetup& iSetup) {
   iEvent.getByToken(hfrechit_token_, TheHFRecHits);
 
   // Run the HcalHaloAlgo to reconstruct the HcalHaloData object
-  HcalHaloAlgo HcalAlgo;
   HcalAlgo.SetRecHitEnergyThresholds(HBRecHitEnergyThreshold, HERecHitEnergyThreshold);
   HcalAlgo.SetPhiWedgeThresholds(SumHcalEnergyThreshold, NHitsHcalThreshold);
 

